frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1297,6 +1305,36 @@ function ReaderBookmark:getBookmarkPageString(page)
     return tostring(page)
 end
 
+function ReaderBookmark:getBookmarkedPages()
+    local pages = {}
+    for i, v in ipairs(self.bookmarks) do
+        local page
+        if self.ui.document.info.has_pages then
+            page = v.page
+        else
+            page = self.ui.document:getPageFromXPointer(v.page)
+        end
+        local btype
```

Comment:
I propose to add `bookmark.text == ""` to `ReaderBookmark:isBookmarkAutoText` and use `ReaderBookmark:getBookmarkType` here.

Reply:
You mean I should add this ?:
```diff
--- a/frontend/apps/reader/modules/readerbookmark.lua
+++ b/frontend/apps/reader/modules/readerbookmark.lua
@@ -1318,3 +1318,3 @@ function ReaderBookmark:getBookmarkedPages()
         if v.highlighted then
-            if v.text == nil or v.text == "" or self:isBookmarkAutoText(v) then
+            if self:isBookmarkAutoText(v) then
                 btype = "highlight"
@@ -1349,3 +1349,3 @@ end
 function ReaderBookmark:isBookmarkAutoText(bookmark)
-    return (bookmark.text == nil) or (bookmark.text == bookmark.notes)
+    return (bookmark.text == nil) or (bookmark.text == "") or (bookmark.text == bookmark.notes)
         or (bookmark.text == self:getBookmarkAutoText(bookmark, true))
```

Sure it's ok in the other contexts you use isBookmarkAutoText() ?
(I don't remember why I ended  up with these 3 checks, while the 1st is done in the current isBookmarkAutoText() - may be to avoid the isBookmarkAutoText() function call in the most common cases...)

Reply:
Yes for replacement in `isBookmarkAutoText`.
Currently "" should not exist in the `text` field, maybe some of the old bookmarks have it.
Anyway, it does not hurt and would be replaced 
https://github.com/koreader/koreader/blob/edd4f0baf74414252082ddf44843dc773eb42c0d/frontend/apps/reader/modules/readerbookmark.lua#L412-L414

And then just `btype = self:getBookmarkType(v)`.

Reply:
Oh, I thought you were proposing that I add ReaderBookmark:getBookmarkType(), which I did, and only later I discovered it was already existing :)

Reply:
Should I also update it on line 955 ?
```diff
--- a/frontend/apps/reader/modules/readerbookmark.lua
+++ b/frontend/apps/reader/modules/readerbookmark.lua
@@ -411,3 +411,3 @@ function ReaderBookmark:onShowBookmark(match_table)
         local v = self.bookmarks[is_reverse_sorting and i or num - i]
-        if v.text == nil or v.text == "" then
+        if self:isBookmarkAutoText(v) then
             v.text = self:getBookmarkAutoText(v)
@@ -954,3 +954,3 @@ function ReaderBookmark:renameBookmark(item, from_highlight, is_new_note)
                 bm.pboxes = pboxes
-                if bm.text == nil or bm.text == "" then
+                if self:isBookmarkAutoText(bm) then
                     bm.text = self:getBookmarkAutoText(bm)
```
(Not willing to think about how to test this, just following your orders :)

Reply:
Ok, done, and simplified things a bit, avoiding a loop:
https://github.com/koreader/koreader/compare/edd4f0baf74414252082ddf44843dc773eb42c0d..b3d84008c8f6fc50c267a02220476a661b0b2853
Is that ok for you?

Reply:
No, 412 and 955 are just initial filling, keep them as is.

Reply:
> which I did, and only later I discovered it was already existing :)

Sorry that I was unclear.

Reply:
> No, 412 and 955 are just initial filling, keep them as is.

OK, done.